sdk/core/core/inc/az_json.h

```diff
@@ -24,28 +24,42 @@
 
 #include <_az_cfg_prefix.h>
 
-/*
- * @brief az_json_token_kind is an enum defining symbols for the various kinds of JSON tokens.
+/**
+ * @brief Defines symbols for the various kinds of JSON tokens that make up any JSON text.
  */
 typedef enum
 {
-  AZ_JSON_TOKEN_NULL,
-  AZ_JSON_TOKEN_BOOLEAN,
-  AZ_JSON_TOKEN_NUMBER,
-  AZ_JSON_TOKEN_STRING,
-  AZ_JSON_TOKEN_OBJECT,
-  AZ_JSON_TOKEN_OBJECT_START,
-  AZ_JSON_TOKEN_OBJECT_END,
-  AZ_JSON_TOKEN_ARRAY_START,
-  AZ_JSON_TOKEN_ARRAY_END,
-  // AZ_JSON_TOKEN_SPAN represents a token consisting of nested JSON; the JSON parser never returns
-  // a token of this type.
-  AZ_JSON_TOKEN_SPAN,
+  AZ_JSON_TOKEN_NONE, ///< There is no value (as distinct from #AZ_JSON_TOKEN_NULL).
+  AZ_JSON_TOKEN_OBJECT_START, ///< The token kind is the start of a JSON object.
+  AZ_JSON_TOKEN_OBJECT_END, ///< The token kind is the end of a JSON object.
+  AZ_JSON_TOKEN_ARRAY_START, ///< The token kind is the start of a JSON array.
+  AZ_JSON_TOKEN_ARRAY_END, ///< The token kind is the end of a JSON array.
+  AZ_JSON_TOKEN_PROPERTY_NAME, ///< The token kind is a JSON property name.
+  AZ_JSON_TOKEN_STRING, ///< The token kind is a JSON string.
+  AZ_JSON_TOKEN_NUMBER, ///< The token kind is a JSON number.
+  AZ_JSON_TOKEN_TRUE, ///< The token kind is the JSON literal `true`.
+  AZ_JSON_TOKEN_FALSE, ///< The token kind is the JSON literal `false`.
+  AZ_JSON_TOKEN_NULL, ///< The token kind is the JSON literal `null`.
 } az_json_token_kind;
 
-/*
+/**
+ * @brief A limited stack used by the #az_json_builder to track state information for validation.
+ */
+typedef struct
+{
+  struct
+  {
+    // This uint64_t container represents a tiny stack to track the state during nested transitions.
+    // The first bit represents the state of the current depth (1 == object, 0 == array).
+    // Each subsequent bit is the parent / containing type (object or array).
+    uint64_t az_json_stack;
+    int32_t current_depth;
+  } _internal;
+} _az_json_bit_stack;
```

Comment:
Sure. Given the other names here are JSON related, I will update it. 

That said, I think a general purpose `bit_stack` (where the fact that is a bit stack is a defining characteristic) like this could potentially be useful in other cases and there isn't a lot that is specific to JSON here (just 0 and 1s).  A similar concept of a `bit array` exists in .NET as well :)

Reply:
Your call. Do whatever you think best.